hs_core/search_indexes.py

```diff
@@ -879,112 +824,122 @@ def prepare_field_type_code(self, obj):
+        Index both variable names and variable codes, due to inconsistent use. 
         """
-        variable_names = set()
+        variables = set()
         for f in obj.logical_files:  # could take a long time if lots of files
             if hasattr(f, "metadata"):
                 if isinstance(f.metadata, NetCDFFileMetaData):
                     for v in f.metadata.variables.all():
                         if discoverable(v.name):
-                            variable_names.add(v.name.strip())
+                            variables.add(v.name.strip())
                 elif isinstance(f.metadata, TimeSeriesFileMetaData):
                     for v in f.metadata.variables:
                         if discoverable(v.variable_name):
-                            variable_names.add(v.variable_name.strip())
+                            variables.add(v.variable_name.strip())
+                        if discoverable(v.variable_code):
+                            variables.add(v.variable_code.strip())
                 elif isinstance(f.metadata, RefTimeseriesFileMetaData):
                     for v in f.metadata.variables:
                         if discoverable(v.name):
-                            variable_names.add(v.name.strip())
+                            variables.add(v.name.strip())
+                        if discoverable(v.code):
+                            variables.add(v.code.strip())
                 elif isinstance(f.metadata, GeoRasterFileMetaData):
                     for b in f.metadata.bandInformations:
                         if discoverable(b.variableName):
-                            variable_names.add(b.variableName)
-        return list(variable_names)
+                            variables.add(b.variableName)
+        return list(variables)
 
     def prepare_variable_type(self, obj):
         """
         Return metadata variable types if exists, otherwise return empty array.
+        Variable type does not exist for referenced time series files.
+        Deprecated. Not particularly useful as a search locator. 
         """
-        variable_types = []
-        if hasattr(obj, 'metadata'):
-            if isinstance(obj.metadata, NetcdfMetaData):
-                for variable in obj.metadata.variables.all():
-                    variable_types.append(variable.type.strip())
-            elif isinstance(obj.metadata, RefTSMetadata):
-                for variable in obj.metadata.variables.all():
-                    variable_types.append(variable.data_type.strip())
-            elif isinstance(obj.metadata, TimeSeriesMetaData):
-                for variable in obj.metadata.variables:
-                    variable_types.append(variable.variable_type.strip())
-        return variable_types
+        variable_types = set()
+        for f in obj.logical_files:  # could take a long time if lots of files
+            if hasattr(f, "metadata"):
+                if isinstance(f.metadata, NetCDFFileMetaData):
```

Comment:
Take a look at the logical_files property implementation.  You can query for the logical file types you need instead of reading them all and checking the type in python. Extracting variable types for netcdf logical files could look like this.
```
for f in obj.netcdflogicalfile_set.all():
    for v in f.metadata.variables.all():
        if discoverable(v.type):
            variable_types.add(v.type.strip())
```
I'm not sure if we can speed it up more than that. I'm not sure how to push more of this logic to the db thanks to the GenericRelation structure we have in the metadata.

All of these methods look to be used for debug printing in a management command and aren't actually used in solar yet?  Am I missing something?

Reply:
This is the actual SOLR ingestion script. The prepare_* routines are called by HayStack to populate SOLR.